ocamltest/run_win32.c

```diff
@@ -370,9 +370,11 @@ int run_command(const command_settings *settings)
     if (timeout != INFINITE)
     {
       cur = GetTickCount();
-      stamp = (cur > stamp ? cur - stamp : MAXDWORD - stamp + cur);
-      timeout = (timeout > stamp ? timeout - stamp : 0);
-      stamp = cur;
+      if (cur > stamp) {
+       DWORD elapsed = cur - stamp;
+       timeout = (timeout > elapsed ? timeout - elapsed : 0);
+       stamp = cur;
+      }
```

Comment:
I am assuming that this is a refactoring. I'm not sure because I am not brave enough this afternoon to read the previous version carefully.

Reply:
As the commit message says, it's a bug fix.  The previous code would set `timeout` to 0 at the first iteration of the loop, causing all commands to timeout almost immediately.  I didn't try to understand what was going wrong in the old code, just rewrote it to do what I think it should do, and voila, timeout is working.

Reply:
Given the propensity of our CI systems to witness random events, the original code was attempting to deal with `GetTickCount` wrapping every other month! I wrote that a few years ago when we still cared about Windows XP - we could just switch to `GetTickCount64` now.

I think `timeout` only becomes zero on the first iteration in the old code if `cur == stamp`... isn't the error just that the first deleted line should be `cur >= stamp`?). I think in your proposed version when `GetTickCount` wraps the timeout becomes effectively `INFINITE` because `stamp` never updates?